contracts/modules/staking/StakingPool.sol

```diff
@@ -1309,7 +1342,9 @@ contract StakingPool is IStakingPool, ERC721 {
       products[_param.productId] = _product;
     }
 
-    require(_totalEffectiveWeight <= MAX_TOTAL_WEIGHT, "StakingPool: Total max effective weight exceeded");
+    if (_totalEffectiveWeight > MAX_TOTAL_WEIGHT) {
+      require(!targetWeightIncreased, "StakingPool: Total max effective weight exceeded");
+    }
```

Comment:
I'd suggest swapping the conditions for `require` and `if` as it feels more natural/expressive:
```
    if (targetWeightIncreased) {
      require(_totalEffectiveWeight <= MAX_TOTAL_WEIGHT, "StakingPool: Total max effective weight exceeded");
    }
```